src/core/state.py

```diff
@@ -1,3 +1,63 @@
-class State():
-    def __init__(self, parameters: int) -> None:
-        self._parameters = parameters
\ No newline at end of file
+import numpy as np
+from typing import Dict
+
+
+class State:
+    """
+    This is a container class for all state variables as defined in this sheet: https://cornell.box.com/s/z20wbp66q0pseqievmadf515ucd971g2.
+    """
+
+    def __init__(self, state_dict: Dict = {}):
+        self.time = 0.0
```

Comment:
Perhaps instead of zeroing all these values, we simply declare them with the following syntax `variable_name: variable_type` (ex: `self.time: float`). This could help us avoid any initialization issues.

Reply:
Idk if this messes up one of your test cases though...

Reply:
I'm setting them to 0.0 now but I think ultimately we want to set them to some other default value that's more meaningful than 0.0, so 0.0 is just a placeholder for now. @tmf97 any thoughts?

Reply:
I think zeros are fine.